pkg/controller/servicebindingrequest/secret.go

```diff
@@ -46,16 +49,33 @@ func (s *secret) createOrUpdate(payload map[string][]byte) (*unstructured.Unstru
 	resourceClient := s.buildResourceClient()
 
 	logger.Debug("Attempt to create secret...")
-	_, err = resourceClient.Create(u, metav1.CreateOptions{})
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return nil, err
-	}
-
-	logger.Debug("Secret already exists, updating contents instead...")
-	_, err = resourceClient.Update(u, metav1.UpdateOptions{})
+	existingSecret, err := resourceClient.Get(s.name, metav1.GetOptions{})
 	if err != nil {
+		if errors.IsNotFound(err) {
+			_, err := resourceClient.Create(u, metav1.CreateOptions{})
+			return u, err
+		}
 		return nil, err
 	}
+	existingSecretData, _, _ := unstructured.NestedMap(existingSecret.Object, "data")
+	existingSecretDataStr := make(map[string]string)
+	for k, v := range existingSecretData {
+		existingSecretDataStr[k] = v.(string)
+	}
+	payloadStr := make(map[string]string)
+	for k, v := range payload {
+		payloadStr[k] = base64.StdEncoding.EncodeToString(v)
+	}
+	eq := reflect.DeepEqual(existingSecretDataStr, payloadStr)
```

Comment:
This transformation can be extracted to its own function, and the comparison instead of using `reflect.DeepEqual` use `cmp.DeepEqual` as we're using in other cases.

Perhaps `nestedMapComparison` can be renamed to `nestedUnstructuredComparison`, and L454-L466 below can be extracted to `nestedMapComparison`:

https://github.com/redhat-developer/service-binding-operator/blob/a7aa2af3a3df9d7afef2f75d26f163155fbc4ff4/pkg/controller/servicebindingrequest/binder.go#L445-L469

Reply:
@isutton here I am comparing between map[string]string , nestedComparison would want a map[string]interface{}.  Payload infact is coming as (payload map[string][]byte) in the createOrUpdate function